google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/apis/ServiceAccountKeyUltimateDisplayDialog.java

```diff
@@ -0,0 +1,303 @@
+    commonPanel = new ServiceAccountKeyDownloadedPanel(gcpProjectId, downloadPath);
+
+    if (tableModel == null) {
+      List<RunnerAndConfigurationSettings> configurationSettingsList =
+          getAppEngineStandardConfigurationSettingsList();
+      tableModel =
+          new BooleanTableModel<>(
+              configurationSettingsList,
+              RunnerAndConfigurationSettings.class,
+              Comparator.comparing(RunnerAndConfigurationSettings::getName),
+              true);
+    }
+    table = new ServerTable(tableModel);
+  }
+
+  private List<RunnerAndConfigurationSettings> getAppEngineStandardConfigurationSettingsList() {
+    if (configurationSettingsList != null) {
+      return configurationSettingsList;
+    }
+
+    return RunManager.getInstance(project)
+        .getConfigurationSettingsList(AppEngineServerConfigurationType.getInstance());
+  }
+
+  private Set<RunnerAndConfigurationSettings> getSelectedConfigurations() {
+    return tableModel.getSelectedItems();
+  }
+
+  @VisibleForTesting
+  public void addEnvironmentVariablesToConfiguration(
+      Set<RunnerAndConfigurationSettings> configurations) {
+    String executorId = DefaultRunExecutor.getRunExecutorInstance().getId();
+    configurations.forEach(
+        configurationSettings ->
+            addEnvironmentVariablesToConfiguration(executorId, configurationSettings));
+  }
+
+  /**
+   * Adds the environment variables for the Google Cloud Libraries to the list of environment
+   * variables for {@code configuration} if they don't exist. If they exist, it replaces them.
+   */
+  private void addEnvironmentVariablesToConfiguration(
+      String executorId, RunnerAndConfigurationSettings configuration) {
+    ProgramRunner runner = ProgramRunnerUtil.getRunner(executorId, configuration);
+    if (runner == null) {
+      LOG.error(
+          "Error updating "
+              + configuration.getName()
+              + " with Google Cloud Library environment variables");
+      return;
+    }
+
+    RunnerSpecificLocalConfigurationBit configurationSettings =
+        (RunnerSpecificLocalConfigurationBit) configuration.getConfigurationSettings(runner);
+    if (configurationSettings == null) {
+      LOG.error(
+          "Error updating "
+              + configuration.getName()
+              + " with Google Cloud Library environment variables");
+      return;
+    }
+
+    Set<EnvironmentVariable> serviceAccountEnvironmentVariables =
+        commonPanel.getEnvironmentVariables();
+    List<EnvironmentVariable> configurationSettingsEnvVariables =
+        configurationSettings.getEnvVariables();
+
+    serviceAccountEnvironmentVariables.forEach(
+        serviceAccountEnvVar -> {
+          OptionalInt indexOpt =
+              IntStream.range(0, configurationSettingsEnvVariables.size())
+                  .filter(
+                      index ->
+                          serviceAccountEnvVar
+                              .getName()
```

Comment:
Do we need to compare them by name here, does `EnvironmentVariable` have any sensible `equals()` implementation? If it does, we may be able to simplify this code a bit.

Reply:
EnvironmentVariable doesn't have equals()